src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -1293,6 +1294,31 @@ let init_from_config_file ?(genesis_dir = Cache_dir.autogen_path) ~logger
                proof_level= Full. The 'proof' field in the configuration file \
                should be removed." ;
             return () )
+          else if
+            (* DO NOT DO THIS: Pretend that the constraint constants match. *)
+            Genesis_constants.Constraint_constants.(equal compiled)
+              { constraint_constants with
+                mainnet=
+                  Genesis_constants.Constraint_constants.compiled.mainnet }
+          then
+            (* TODO(mrmr1993): This is morally wrong.
+               The proving keys that this binary is distributed with are
+               useless for this configuration, and the daemon + subprocesses
+               will instead download or generate the multi-GB keys they need on
+               startup.  They'll also be downloaded to /tmp, so if the machine
+               containing them restarts or /tmp is cleared then they will need
+               to be re-downloaded.
+               There are other subtleties too, around stored data. Toggling the
+               mainnet flag here should be done alongside changes to the
+               version number etc. that cause all of this data to be purged
+               instead of loading it into the binary in this configuration.
+
+               Do not take this as an indication that we should do this for
+               other things. We shouldn't. This case is exceptional.
+               This should be removed and replaced with a proper solution as
+               soon as possible.
+              *)
+            return ()
```

Comment:
@imeckler or someone else @mrmr1993 nominates to verify this is okay

Reply:
Also, quick question: Can we make it so we default to hitting this case where we need to download the keys to /tmp when we're not on mainet. That is: In production, 

Can you change whatever config.mlh is necessary to make that happen? My understanding is that you we switch `mainnet true`?

Reply:
I don't think this condition will be hit on testnets or mainnet given we don't specify `proof_level` in the config file (`generated_constraint_constants` will be false). On line 1260 we should check for any constraint configuration being set in the config file and not just proof_level, right?

Reply:
Nvm my previous comment, I misread the `proof` object  as `proof_level` on line 1260.

Reply:
> Can you change whatever config.mlh is necessary to make that happen?

This is not a change to be made lightly. We need to make sure that all of binaries that we package and distribute have a config file with the `{"proof": {"mainnet": false}}` to make sure that we're not creating testnets etc. where people are accidentally revealing valid mainnet signatures.

I propose doing this separately, and with coordination from @lk86, @0x0I.